crypto/bindings/conversion.ts

```diff
@@ -526,6 +653,80 @@ function createRustConversion(wasm: wasm) {
       return mlRo;
     }
 
+    function commitmentsToRust(
+      commitments: ProverCommitments
+    ): WasmProverCommitments {
+      let wComm = self.polyCommsToRust(commitments[1]);
+      let zComm = self.polyCommToRust(commitments[2]);
+      let tComm = self.polyCommToRust(commitments[3]);
+      let lookup = MlOption.mapFrom(commitments[4], lookupCommitmentsToRust);
+      return new ProverCommitments(wComm, zComm, tComm, lookup);
+    }
+    function commitmentsFromRust(
+      commitments: WasmProverCommitments
+    ): ProverCommitments {
+      let wComm = self.polyCommsFromRust(commitments.w_comm);
+      let zComm = self.polyCommFromRust(commitments.z_comm);
+      let tComm = self.polyCommFromRust(commitments.t_comm);
+      let lookup = MlOption.mapTo(
+        commitments.lookup,
+        lookupCommitmentsFromRust
+      );
+      commitments.free();
+      return [0, wComm as MlTupleN<PolyComm, 15>, zComm, tComm, lookup];
+    }
+
+    function lookupCommitmentsToRust(
+      lookup: LookupCommitments
+    ): WasmLookupCommitments {
+      let sorted = self.polyCommsToRust(lookup[1]);
+      let aggreg = self.polyCommToRust(lookup[2]);
+      let runtime = MlOption.mapFrom(lookup[3], self.polyCommToRust);
+      return new LookupCommitments(sorted, aggreg, runtime);
+    }
+    function lookupCommitmentsFromRust(
+      lookup: WasmLookupCommitments
+    ): LookupCommitments {
+      let sorted = self.polyCommsFromRust(lookup.sorted);
+      let aggreg = self.polyCommFromRust(lookup.aggreg);
+      let runtime = MlOption.mapTo(lookup.runtime, self.polyCommFromRust);
+      lookup.free();
+      return [0, sorted, aggreg, runtime];
+    }
+
+    function openingProofToRust(proof: OpeningProof): WasmOpeningProof {
+      let [_, [, ...lr], delta, z1, z2, sg] = proof;
+      // We pass l and r as separate vectors over the FFI
+      let l: MlArray<OrInfinity> = [0];
+      let r: MlArray<OrInfinity> = [0];
+      for (let [, li, ri] of lr) {
+        l.push(li);
+        r.push(ri);
+      }
+      return new OpeningProof(
+        self.pointsToRust(l),
+        self.pointsToRust(r),
+        self.pointToRust(delta),
+        fieldToRust(z1),
+        fieldToRust(z2),
+        self.pointToRust(sg)
+      );
+    }
+    function openingProofFromRust(proof: WasmOpeningProof): OpeningProof {
+      let [, ...l] = self.pointsFromRust(proof.lr_0);
```

Comment:
IIUC, it ignores the first element of the list, i.e. the `0` for the Caml array memory repr?

Reply:
yes!